pkg/authenticator/authenticator.go

```diff
@@ -254,6 +263,30 @@ func (auth *Authenticator) ParseAuthenticationResponse(response []byte) error {
 	return nil
 }
 
+func waitForCertificate(certificatePath string, timeout time.Duration) error {
+	deadline := time.Now().Add(timeout)
+	for {
+		info, err := os.Stat(certificatePath)
+
+		if !os.IsNotExist(err) && !info.IsDir() {
+			// Certificate file is available
+			log.InfoLogger.Printf(log.CAKC017I, certificatePath)
+
+			// No error, the certificate exists within the deadline
+			return nil
+		}
+
+		if time.Now().After(deadline) {
+			// Timed out waiting for the certificate to exist
+			return log.RecordedError(log.CAKC033E, certificatePath)
```

Comment:
ah, i see see here that you recorded it. never mind my previous comment on this.